src/components/PeopleList/PeopleList.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useContext } from 'react';
+import { useLocation } from 'react-router-dom';
+import { PeopleContext } from '../../PeopleContext';
+import { PeopleItem } from '../PeopleItem/PeopleItem';
+import { Person } from '../../types';
+import { Sex } from '../../enum';
+
+export const PeopleList: React.FC = () => {
+  const { persons } = useContext(PeopleContext);
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+
+  const sex = searchParams.get('sex');
+  const query = searchParams.get('query');
+  const centuries = searchParams.getAll('centuries').map(Number);
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+
+  const filterCondition = (person: Person, filterSex: string) => {
+    return person.sex === filterSex && person.name.toLowerCase()
+      .includes(query?.toLowerCase() || '');
+  };
+
+  const centFilter = () => {
+    if (centuries.length > 0) {
+      return persons.filter((person) => {
+        return centuries.includes(Math.ceil(person.born / 100));
+      });
+    }
+
+    return persons;
+  };
+
+  const tableSort = () => {
+    if (sort === 'name' || sort === 'sex') {
```

Comment:
name and sex are a magic words.  Consider will be better if you create a enum for magic words and use it everywhere